Research: 

Research Jobs | Position | Chances | Oppurtunity: Research and Academic Roles as Bioinformatician, Computational Biology and Genomics, NGS Specialist, Research Scientist, Bio-Software Developer, Bioinformatics CTO, and Lead researcher roles as Senior CTO, Senior Computational Lead and Bioinformatician/Associate. I am not interested in freelancing and technician (so please kindly don't contact regarding the same). 

Genomic analysis and software development for illumina and long read sequencing approaches. I provide bio-software development, algorithms, bioinformatics, machine learning for Illumina and LongRead sequencing technologies across all species. What i do and what i have done is outlaid below including classification of the programming languages for each task, so that you can have a broader idea of what you are looking into and what i extend. I am specific to the analysis in the areas of RNA-seq and Single cell, Metagenomics and Pangenomics covering all species. Similarly, Biological Machine and Deep learning specific to biological questions, biological language models and bio-software. 

Research and Topics and areas, I dont work, read and explore: System Design and Management, GeoInformatics or Machine Learning for GeoInformatics. I dont do LLMs and RAG based fine tuning models for text analysis or chatbots. I only use machine learning and deep learning for biological sciences. I am not a programming lnaguage teacher and not interested in such freelancing roles. I only read and write, RUST, Python, R and Golang. 

Sequencing specifics: \
2010-2021: Plant, Bacterial, Fungi: RNASeq, GenomSeq, Phylogenomics, PacBio Sequencing, Single Cell Analysis, Bio-software. \
2021:2023: Machine Learning, Bio-software. \
2024: PanGenome, Bio-software. \
2025: Human Genomics, Bio-software \
Software Development:C++(2010-2021), 2024-: RUST \
Web Development: RUST \
Bioinformatics/Data Analysis/Machine and Deep Learning: Python or R.\
HPC Management: Golang(2024:-).
RUST release: See the commit tag as final or build release for the release.

Background:

I have a background in bioinformatics and plant biology and extensive experience in code development using several programming languages(prior to 2021 - C++, Bash, R - after 2021: Bash, Python,Golang, RUST) covering data analysis and data science, machine and deep learning and web and application development. Following my PhD, I developed bioinformatics methods for transcriptional and post-transcriptional genomics across nuclear and organelle genomes at Fondazione Edmund Mach (Italy). I analyzed and finished multiple RNA-seq and Organelle-Seq experiments for several plant and fungal species, including Arundo donax. Additionally i analyzed multiple metagenomics anlaysis coming from the fungal and bacterial species involving ITS metagenomics, as well as the bacterial metagenomics.

Additionally, I have done a lot of work in the field of organelle genomics and have published the first Cardamine species' chloroplast genomes. I independently created an international partnership to find and create computational methods for a number of crop species. Following that, I spent two years (2014â€“2016) as a Research Fellow (Academic Level B) at the University of Technology, Sydney, Australia, where I developed computational methods for understanding seagrasses. Following that, I spent a short time spent in University of Connecticut, USA, where i analyzed the Douglas fir genome from the genome annotation to the phylogenomics and identifying genes and evolution of importance.

Since August 2017, I have worked as a Postdoctoral researcher at the Finnish Museum of Natural History and the University of Helsinki, conducting research on genome bioinformatics and sequencing the genomes of lower plants, including Coleochaete orbicularis, Blasia pusilla, Chaetospiridium orbicularis, Polytrichum commune, Mallomonas, and Cryptomonas species. My work has been focused on genome assembly, genome annotation, chloroplast genomics, and a variety of other topics. Additionally, I've worked for various other organisations, such as Edinburgh UK, to analyse the genomics data for PAFTOL species and the chloroplast genomes of the Ambrosia clade from Norway.

Since 2019, my research has shifted to examining the genomes of fungi whose species have been sequenced using NextSeq methods. This work is currently concentrated on genome assembly, annotations, markers genes, and phylogenomics of those fungi. I have assembled, annotated, and identified ITS and other phylogenomics markers, as well as performed alignments, phylogenies, and downstream analyses on the fungal genomes of over 500 different species. The bioinformatics application of high throughput sequencing and methods to comprehend the biological and functional importance of the genes, evolution, and pathways in plants have been the main areas of my research up to this point.

From 2022-2023, i added several new skills as a carrier advancement. From 2024 onwards, I worked at Universitat Potsdam, Germany, where I self-learnt RUST and develop approaches for machine and deep learning. During the time, I bench-marked PacBioHifi genome analysis and created a complete HMTL, CSS, Javascript enabled web and also coded several approaches, packages in Python and  RUST.  Since 2025, I worked as Area Expert at Instytut Chemii Bioorganicznej Polskiej Akademii Nauk, Poland, and i worked on human genomics and developed computational approaches for human genomics and software development. 

 
