Research: 

**Research and Academic Roles** as Bioinformatician, Computational Biology and Genomics, NGS Specialist, Research Scientist, Bio-Software Developer, Bioinformatics CTO, and Lead researcher roles as Senior CTO, Senior Computational Lead and Bioinformatician/Associate. I am not interested in freelancing and technician jobs (so please kindly don't contact regarding the same). Genomic analysis and software development for illumina and long read sequencing approaches. I provide bio-software development, algorithms, bioinformatics, machine learning for Illumina and LongRead sequencing technologies across all species. I am specific to the analysis in the areas of RNA-seq and Single cell, Metagenomics and Pangenomics covering all species. Similarly, Biological Machine and Deep learning specific to biological questions, biological language models and bio-software, HPC management. I prefer academic and research areas in University and Organization. 

**Academic and Research teaching**: I only read and write, fullstack Python and RUST for the scientific and academic research. As a part of the scientific and academic research, I can give the language classes to academics and research students. I am not a private teacher or YouTuber and not interested in such freelancing roles or forming collaboration on the same. Read the software research for more detail so that my thinking can be gleaned. 

**Sequencing specifics**: \
2010-2021: Plant, Bacterial, Fungi: RNASeq, GenomSeq, Phylogenomics, PacBio Sequencing, Single Cell Analysis, Bio-software. \
2021:2023: Machine Learning, Bio-software. \
2024: PanGenome, Bio-software. \
2025: Human Genomics, Bio-software \
**Software Development:** C++(2010-2021), 2024-: RUST \
**Web Development:** RUST ðŸ¦€ i use Axum, Rocket, Dioxus, Actix, Leptos web framework and Python\
**Bioinformatics/Machine and Deep Learning:** RUST and Python using Keras, XgBoost, Tensorflow and PyTorch. \
**HPC Management:** RUST and Python (2024:-). 

**Background:**

I have a background in bioinformatics and plant biology and extensive experience in code development using several programming languages(prior to 2021 - C++, Bash, R - after 2021: Bash, Python, RUST) covering data analysis and data science, machine and deep learning and web and application development. Following my PhD, I developed bioinformatics methods for transcriptional and post-transcriptional genomics across nuclear and organelle genomes at Fondazione Edmund Mach (Italy). I analyzed and finished multiple RNA-seq and Organelle-Seq experiments for several plant and fungal species, including Arundo donax. Additionally i analyzed multiple metagenomics anlaysis coming from the fungal and bacterial species involving ITS metagenomics, as well as the bacterial metagenomics.Additionally, I have done a lot of work in the field of organelle genomics and have published the first Cardamine species' chloroplast genomes. I independently created an international partnership to find and create computational methods for a number of crop species. Following that, I spent two years (2014â€“2016) as a Research Fellow (Academic Level B) at the University of Technology, Sydney, Australia, where I developed computational methods for understanding seagrasses. Following that, I spent a short time spent in University of Connecticut, USA, where i analyzed the Douglas fir genome from the genome annotation to the phylogenomics and identifying genes and evolution of importance.

Since August 2017, I have worked as a Postdoctoral researcher at the Finnish Museum of Natural History and the University of Helsinki, conducting research on genome bioinformatics and sequencing the genomes of lower plants, including Coleochaete orbicularis, Blasia pusilla, Chaetospiridium orbicularis, Polytrichum commune, Mallomonas, and Cryptomonas species. My work has been focused on genome assembly, genome annotation, chloroplast genomics, and a variety of other topics. Additionally, I've worked for various other organisations, such as Edinburgh UK, to analyse the genomics data for PAFTOL species and the chloroplast genomes of the Ambrosia clade from Norway. Since 2019, my research has shifted to examining the genomes of fungi whose species have been sequenced using NextSeq methods. This work is currently concentrated on genome assembly, annotations, markers genes, and phylogenomics of those fungi. I have assembled, annotated, and identified ITS and other phylogenomics markers, as well as performed alignments, phylogenies, and downstream analyses on the fungal genomes of over 500 different species. The bioinformatics application of high throughput sequencing and methods to comprehend the biological and functional importance of the genes, evolution, and pathways in plants have been the main areas of my research up to this point.

From 2022-2023, i added several new skills as a carrier advancement. From 2024 onwards, I worked at Universitat Potsdam, Germany, where I self-learnt RUST and develop approaches for machine and deep learning. During the time, I bench-marked PacBioHifi genome analysis and created a complete HMTL, CSS, Javascript enabled web and also coded several approaches, packages in Python and  RUST.  Since 2025, I worked as Area Expert at Instytut Chemii Bioorganicznej Polskiej Akademii Nauk, Poland, and i worked on human genomics and developed computational approaches for human genomics and software development. 

**Software Research**: Over the years, I have worked with several languages such as C++ (2010-2021), R (2010-2025), Python (2021-), GOlang (2024-2025), RUST (2024-) but mostly used one system programming language such as C++(2010-2021) and then replaced with RUST(2024-) and R (2010-) replaced with Python (2021-). I still use R as a part of the analysis but mostly where it is not having a Python companion. I believe more in addressing software development using the programming languages i prefer and not to show off multiple languages. Hence i take the languages, which are dominant in their field. 

As a fantatic reader and coder (I wrote and developed every software as single lead bioinformatician at every employment), I sometimes read a little bit about other languages but that doesnt mean, I plan to use them. I do this to enhance my knowledge and bring that features to the language i use. As of now, I only code in RUST as a full stack developer and Python for machine learning and interpolation with RUST. I believe a Bioinformatician/Software Developer should be well informed of the software research in their areas so that they can actively use them for the place of the employment. 

Use of selected languages for everything has enabled me to scale my abilities to everything at every employment and i served as a single lead bioinformatician from software developer to bioinformatics data analysis, machine learning and HPC cloud management. 


**Python and RUST development**: Few of them and click the source for more of them. See the last commit tag as the final build release for each of the source repository. Over 5K crates download. **These languages I only plan to use forever for anything in my software developer for bioinformatics, machine and deep learning, HPC and cloud management.**

-> [eVaiUtilities](https://github.com/omicscode/eVaiutilities): Variant analysis from the eVai. \
-> [panscape](https://github.com/omicscode/panscape): Pangenome long reads \
-> [sequenceprofiler](https://github.com/omicscode/sequenceprofiler): Profiling sequence kmers for histograms \
-> [phyloevolve](https://github.com/omicscode/phyloevolve): Long reads and alignments from the multiple alignments. \
-> [hpcMapper](https://github.com/omicscode/hpcMapper): DevOPS system managment for the high performance computing.\
-> [bacdive](https://github.com/omicscode/bacdive): Bacterial genome analysis from Bacdive.\
-> [NLRanalyzer](https://github.com/omicscode/NLRanalyzer): Complete kit for analyzing NLR.\
-> [humanCAST](https://github.com/omicscode/humanCAST): Complete kit for human genome analysis.\
-> [araseq](https://github.com/omicscode/araseq): Complete kit for the Arabidopsis genome information.\
-> [minifyseq](https://github.com/omicscode/minifySeq): Noise removal from the long reads inclduing the machine learning based.\
-> [CAGanalyzer](https://github.com/omicscode/CAGrepeat-analyzer): Analyzing the CAG repeats from the human genome.\
-> [doiTAG](https://github.com/omicscode/doiTAG): Generating doi for the sequences for next generation sequencing.\
-> [varLinker](https://github.com/omicscode/varlinker): Analyzing and linking variants for annotation.\
-> [vcfFilter](https://github.com/omicscode/vcfilter): Filtering VCF files\
-> [rustRet](https://github.com/omicscode/rustRet): Analyzing the massspectrometry data.\
-> [repgnerate](https://github.com/omicscode/repgenerate): Analyzing the sequencing information post sequencing.\
-> [proteogenomics](https://github.com/omicscode/proteogenomics): implementing the proteogenomics methods.\
-> [geomapper](https://github.com/omicscode/geomapper): a complete kit for geospatial analysis for German geo.\
-> [ensemblcov](https://github.com/omicscode/ensemblcov): a complete kit for using the ensemblcov at your commandline.\
-> [bactiPAN](https://github.com/omicscode/varView): complete bacterial pangenome analyzer both in shell and RUST.\
-> [varView](https://github.com/omicscode/varView): Graphics enabled Variant terminal analyzer in GOlang.\
-> [doseqGO](https://github.com/omicscode/doseqGO): A complete sequence information portal for standalone sequence information. 

 
